tests/e2e/e2e.go

```diff
@@ -0,0 +1,442 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ava-labs/avalanche-network-runner/rpcpb"
+	"github.com/ava-labs/avalanchego/config"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/rpc"
+	"github.com/ava-labs/hypersdk/tests/workload"
+	"github.com/ava-labs/hypersdk/utils"
+
+	runner_sdk "github.com/ava-labs/avalanche-network-runner/client"
+	ginkgo "github.com/onsi/ginkgo/v2"
+)
+
+const (
+	modeTest = "test"
+	modeRun  = "run"
+)
+
+var (
+	networkConfig     Config
+	network           *embeddedVMNetwork
+	uris              []string
+	txWorkloadFactory workload.TxWorkloadFactory
+	anrCli            runner_sdk.Client
+	trackSubnetsOpt   runner_sdk.OpOption
+	mode              string
+	blockchainID      string
+)
+
+func SetWorkloadFactory(
+	factory workload.TxWorkloadFactory,
+) {
+	txWorkloadFactory = factory
+}
+
+type embeddedVMNetwork struct {
+	uris      []string
+	networkID uint32
+	chainID   ids.ID
+}
+
+type Config struct {
+	Name                       string
+	Mode                       string
+	ExecPath                   string
+	NumValidators              uint
+	GRPCEp                     string
+	NetworkRunnerLogLevel      string
+	AvalanchegoLogLevel        string
+	AvalanchegoLogDisplayLevel string
+	VMGenesisPath              string
+	VMConfig                   string
+	SubnetConfigPath           string
+	PluginDir                  string
+}
+
+func CreateE2ENetwork(
+	ctx context.Context,
+	c Config,
+) {
+	networkConfig = c
```

Comment:
Hmm this is a bit of a strange case because the entry point provides the workload, but it's actually importing the tests themselves

Reply:
We could separate the configuration into a separate package from the tests, but the tests will just depend on that package to configure the network, so I'm not sure that helps much.

We could move network setup and teardown into a separate package, so that the entry point then imports one package for network construction/teardown and a separate package for pre-defined test coverage from the HyperSDK. I'm not sure this is necessary as I think this is likely the only pre-defined test coverage we'll want to add.